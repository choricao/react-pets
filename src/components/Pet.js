import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Pet extends Component {

  render() {
    return (
      <article>
        <h3>Pet {this.props.name}</h3>
        <p><strong>Age: </strong> {this.props.age}</p>
        <p><strong>Breed: </strong> {this.props.breed}</p>
        <p>{this.props.about}</p>
      </article>
    );
  }
}

Pet.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  breed: PropTypes.string.isRequired,
  about: PropTypes.string.isRequired,
};

export default Pet;
