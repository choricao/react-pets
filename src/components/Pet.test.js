import React from 'react';
import Pet from './Pet';
import { mount } from 'enzyme';

describe('Pet', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = mount(<Pet />);

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();

    // Remove the component from the DOM (save memory and prevent side effects).
    wrapper.unmount();
  });
});
