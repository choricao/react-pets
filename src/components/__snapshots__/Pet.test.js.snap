// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pet that it matches an existing snapshot 1`] = `
<Pet>
  <article>
    <h3>
      Pet 
    </h3>
    <p>
      <strong>
        Age: 
      </strong>
       
    </p>
    <p>
      <strong>
        Breed: 
      </strong>
       
    </p>
    <p />
  </article>
</Pet>
`;
